doc/userguide/src/ExecutingTestCases/ConfiguringExecution.rst

```diff
@@ -31,11 +31,21 @@ extensions, the :option:`--extension (-F)` option must be used to explicitly
 tell the framework to parse also them. If there is a need to parse more
 than one kind of files, it is possible to use a colon `:` to separate
 extensions. Matching extensions is case insensitive and the leading `.`
-can be omitted::
+can be omitted. You can additionally filter files before parsing using
+the :option:`--files (-f)` option with a `simple pattern`_. The option
+can be used multiple times to match multiple patterns. Arguments to the
+:option:`--files (-f)` option are case- and space-insensitive.
+If the :option:`--files (-f)` option matches files with a non-default extension,
+the :option:`--extension (-F)` option must be added in order for those files
+to also be parsed.
+::
```

Comment:
Explaining `--extension` and `--files` in the same section is a good idea. Documentation of `--files` should get its own paragraph, though. Possibly there could also be separate subsections.

Another thing that's needed is a note about `--files` being new in RF 6.1. There can either be an explicit, `.. note` or the paragraph explaining `--files` can start with something like "Starting from Robot Framework 6.1 ...". There ought to be examples of both styles here.